### PR TITLE
Make it possible to do a Guided Transfer to Pressable.

### PR DESCRIPTION
**Pull request by dllh**
We've been working on adding Guided Transfer to Calypso generally, and are ready for test of that. Before we test, I'd like to get Pressable transfers into Calypso as well, so that we can test all the use cases we hope to ship with.

The code here is based on code that has already been reviewed and duplicates patterns and components in use for the hosts that have already landed, so from a code review perspective, this should be pretty simple.

**To test:** 

1. Set up an account and a test site at pressable.com, with a domain.
2. In Calypso, go to My Sites and choose a site for which you have no Guided Transfer subscription.
3. Now navigate to Settings -> Export and choose the Guided Transfer option.
4. Select Pressable as the host.
5. Enter credentials and the domain you wish to transfer to in your Pressable account.
6. Once credentials have been validated, complete the purchase (consider giving yourself a coupon to ease the testing process)

**What to expect**
You should get an email letting you know that the purchase is complete and another letting you know that the GT will be started.

After a few minutes, the transfer should complete automatically, but you can check with GT HEs or check the GT queue for your transfer to see details.

**Things to test or consider**
- Supply incorrect hosting credentials.
- Try buying a transfer for a site that already had a transfer subscription.
- Try initiating a transfer for a site with a premium theme or custom fonts.
- Does the email make sense?
- Does the transfer work?

**Screen Shots**

Export Screen:
screen shot 2016-10-12 at 2 55 31 pm

Host Selection:
screen shot 2016-10-12 at 2 55 42 pm

Credential Entry:
screen shot 2016-10-12 at 2 55 50 pm

Post-purchase Screen:
screen shot 2016-10-12 at 2 58 15 pm

### CODE REVIEW

**Comment by dllh** [comment]
Cc @jordwest for review.

File: client/my-sites/guided-transfer/host-credentials-page/pressable.jsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+						hostUrl={ hostInfo.url }
+					/>
+					<WPOrgURL
+						value={ fieldValues.wporg_url }
+						onChange={ onFieldChange( 'wporg_url' ) }
+						disabled={ isSubmitting }
+					/>
+				</CompactCard>
+				<SubmitSection
+					submit={ this.props.submit }
```
Comment (omarjackman) [inline client/my-sites/guided-transfer/host-credentials-page/pressable.jsx]: Not a big deal but you could also add `submit` to your destructuring assignment on [line 28](https://github.com/Automattic/wp-calypso/pull/8170/files#diff-7be0888d3ad63f44695bd14a911dc318R28)

File: client/my-sites/guided-transfer/guided-transfer.jsx [hunk 2]
```diff
@@ -26,6 +26,11 @@ const guidedTransferHosts = {
 		logo: '/calypso/images/guided-transfer/siteground-logo.png',
 		label: i18n.translate( 'SiteGround' ),
 		url: 'https://www.siteground.com/wordpress-hosting.htm?afcode=134c903505c0a2296bd25772edebf669'
+	},
+	pressable: {
+		logo: '/calypso/images/guided-transfer/pressable-logo.png',
+		label: i18n.translate( 'Pressable' ),
+		url: 'https://pressable.com'
```
Comment (omarjackman) [inline client/my-sites/guided-transfer/guided-transfer.jsx]: Is there any URL we could use with pressable that includes tracking?

File: client/my-sites/guided-transfer/host-credentials-page/pressable.jsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+import CompactCard from 'components/card/compact';
+import {
+	Email,
+	Password,
+	CreateAccountTip,
+	SubmitSection,
+	WPOrgURL,
+} from './fields';
+
+class Pressable extends Component {
```
Comment (omarjackman) [inline client/my-sites/guided-transfer/host-credentials-page/pressable.jsx]: I know you didn't introduce this but I think these `host-credentials-page` components could/should be stateless functions instead of classes. From what I've read React plans to add performance optimizations for stateless functions that we'll be able to take advantage of.

Comment (jordwest) [inline client/my-sites/guided-transfer/host-credentials-page/pressable.jsx]: Agreed this makes sense to be a stateless functional component. Don't think it's a blocker here though, let's do it in a separate PR along with the Bluehost and SiteGround forms

**Comment by jordwest** [comment]
Looks good to me. The asymmetry of the button alignment looks a little weird though. Perhaps in a followup PR we could put the Pressable button at the top, fill the horizontal space making it bigger and more..... pressable :trollface:

File: client/my-sites/guided-transfer/guided-transfer.jsx (hunk 2 above)
Comment (dllh) [inline client/my-sites/guided-transfer/guided-transfer.jsx]: Yeah, there may be another url, but I'd rather not hold this up for it.
